### PR TITLE
[Box] Add border width support

### PR DESCRIPTION
**Pull request by aveline**
### WHY are these changes introduced?

Part of [#7601](https://github.com/Shopify/polaris/issues/7601)

### WHAT is this pull request doing?

- Adds border with support only with this PR as further exploration is needed to implement border color
- Sets the initial border width to `--p-border-width-1` which works currently as all the existing shorthand border properties use that width. Setting to `initial` doesn't work as the browser initial is `medium` which overrides the shorthand width.

<details>
<summary>Playground code</summary>

```jsx
      <AlphaStack fullWidth>
        <Box
          paddingInlineEnd="10"
          paddingInlineStart="10"
          border="divider"
          borderWidth="1"
        >
          test
        </Box>
        <Box border="dark" borderWidth="3">
          test
        </Box>
        <Box border="dark" borderBlockStartWidth="3">
          test
        </Box>
        <Box border="dark" borderBlockEndWidth="3">
          test
        </Box>
        <Box border="dark" borderInlineStartWidth="3">
          test
        </Box>
        <Box border="dark" borderInlineEndWidth="3">
          test
        </Box>
        <Box border="dark" borderBlockEndWidth="2">
          test
        </Box>
        <Box
          border="divider"
          borderBlockEndColor="border"
          paddingInlineEnd="10"
          paddingInlineStart="10"
        >
          test
        </Box>
      </AlphaStack>
```

</details>

### CODE REVIEW

File: polaris-react/src/components/Box/Box.tsx [hunk 1]
```diff
@@ -12,72 +18,6 @@ type Element = 'div' | 'span' | 'section';
 
 type Overflow = 'hidden' | 'scroll';
 
-export type BackgroundColorTokenScale =
```
Comment (kyledurand) [inline polaris-react/src/components/Box/Box.tsx:15]: Thank you for moving these ❤️

File: .changeset/gorgeous-dryers-wash.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@shopify/polaris': minor
+'@shopify/polaris-tokens': minor
+---
+
+- Add border width prop to `Box`
+- Export color token subset alias types from tokens package and remove from `Box`
```
Comment (kyledurand) [inline .changeset/gorgeous-dryers-wash.md]: I don't know if we're still trying to do past tense with these but I think it makes more sense in the changelog 🤷 

```suggestion
- Added border width prop to `Box`
- Exported color token subset alias types from tokens package and remove from `Box`
```

File: polaris-react/src/components/Box/Box.tsx [hunk 3]
```diff
@@ -121,6 +61,13 @@ export type BorderRadiusTokenScale =
   | 'large'
   | 'half';
 
+export type BackgroundColors =
+  | ColorsBackdropTokenAlias
+  | ColorsBackgroundTokenAlias
+  | ColorsOverlayTokenAlias
+  | ColorsActionTokenAlias
+  | ColorsSurfaceTokenAlias;
```
Comment (kyledurand) [inline polaris-react/src/components/Box/Box.tsx:69]: Thoughts on this group being exported from tokens as well? Not sure if we'll need it elsewhere

File: polaris-react/src/components/Box/Box.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import type {ComponentMeta} from '@storybook/react';
+import {Box, Icon} from '@shopify/polaris';
+import {PaintBrushMajor} from '@shopify/polaris-icons';
+
+export default {
+  component: Box,
+} as ComponentMeta<typeof Box>;
+
+export function Default() {
```
Comment (kyledurand) [inline polaris-react/src/components/Box/Box.stories.tsx:10]: Do we need a default example if this isn't really a default? Default wouldn't render anything visual except children

**Review by kyledurand — APPROVED** [review]
Nice clean up 👍

**Comment by laurkim** [comment]
The `get-props` script should probably be run with this pr to pick up the new props in the style guide.

**Review by laurkim — APPROVED** [review]
Nice! 💯 
Small nit, the `get-props` script should probably be run with this pr to pick up the new props in the style guide.

File: polaris-react/src/components/Box/Box.tsx (hunk 3 above)
Comment (aveline) [reply polaris-react/src/components/Box/Box.tsx:69]: This seemed like a subset of background colors specific to `Box` which is why I left them in here. But if we find ourselves needing it elsewhere then yes definitely would make sense to export from tokens.

File: polaris-react/src/components/Box/Box.stories.tsx (hunk 4 above)
Comment (aveline) [reply polaris-react/src/components/Box/Box.stories.tsx:10]: Storybook complains if there's no `Default`. But if you know a workaround, I agree this `Box` default is awkward. Maybe I should remove the background from Default and let it render nothing?

Comment (kyledurand) [reply polaris-react/src/components/Box/Box.stories.tsx:10]: Ah right, yeah maybe let's just render the children and show that there's nothing visual that renders when no props are provided 👍

**Pull request merged**